### PR TITLE
Txs patches

### PR DESCRIPTION
**Pull request by nickeskov**
This pull request introduces several important improvements and fixes related to Ethereum signature handling, cryptographic verification, and protobuf order conversion, as well as some general code quality enhancements. The main focus is on correcting how padded Ethereum signatures are processed, improving signature verification robustness, and ensuring compatibility with legacy and current systems.

**Ethereum signature handling and bugfixes:**

* Added `transformToStandardSig` to process and normalize padded Ethereum signatures (129 bytes with extra prefix data) to the standard 65-byte format, fixing a bug in protobuf order conversion and addressing legacy compatibility issues. (`pkg/proto/protobuf_converters.go`, `pkg/proto/types.go`, `pkg/proto/transactions_with_proofs.go`) [[1]](diffhunk://#diff-39cc270751d0512791f183a74d580f49cc6a7cf557d2b1b53c3e7ccf4efd801cL690-R715) [[2]](diffhunk://#diff-39cc270751d0512791f183a74d580f49cc6a7cf557d2b1b53c3e7ccf4efd801cL778-R801) [[3]](diffhunk://#diff-98034364ee83337e7fb1e32bfcef6bb0df53659e6a313f2623c9287c615ae462R1923) [[4]](diffhunk://#diff-98034364ee83337e7fb1e32bfcef6bb0df53659e6a313f2623c9287c615ae462R1995-R2005) [[5]](diffhunk://#diff-deedb90c182198d73a01899c5c676912a79932c9e25f121d22607f92bb63eee6R1842) [[6]](diffhunk://#diff-83b634af2d32736559239fb2491f4bb9ebee7ddb67bd0a122d307fd95b45861cR784-R807)
* Refactored Ethereum signature validation: split out `validateEthereumSignatureRS` for internal use and improved error handling in public key recovery. (`pkg/proto/eth_crypto.go`) [[1]](diffhunk://#diff-0f1c0c4cc6c543505cf001a94c5e127bef2061ad43048da061d79f81090b6f9fL17-R18) [[2]](diffhunk://#diff-0f1c0c4cc6c543505cf001a94c5e127bef2061ad43048da061d79f81090b6f9fL29-R34) [[3]](diffhunk://#diff-0f1c0c4cc6c543505cf001a94c5e127bef2061ad43048da061d79f81090b6f9fL52-R64) [[4]](diffhunk://#diff-0f1c0c4cc6c543505cf001a94c5e127bef2061ad43048da061d79f81090b6f9fL124-R130) [[5]](diffhunk://#diff-0f1c0c4cc6c543505cf001a94c5e127bef2061ad43048da061d79f81090b6f9fR151-R153)
* Standardized signature length constants and updated all usages to `EthereumSignatureLength` for clarity and maintainability. (`pkg/proto/eth_signer.go`, `pkg/proto/eth_crypto.go`) [[1]](diffhunk://#diff-4e61ae67337e881cc82587134df3666886d2e69f2f6926552d1998da438a7dc2R11-R20) [[2]](diffhunk://#diff-4e61ae67337e881cc82587134df3666886d2e69f2f6926552d1998da438a7dc2L498-R502) [[3]](diffhunk://#diff-0f1c0c4cc6c543505cf001a94c5e127bef2061ad43048da061d79f81090b6f9fL52-R64) [[4]](diffhunk://#diff-0f1c0c4cc6c543505cf001a94c5e127bef2061ad43048da061d79f81090b6f9fL124-R130)

**Ed25519 signature verification improvements:**

* Enhanced Ed25519 verification to handle non-canonical `s` values according to the XEdDSA spec, increasing robustness and compatibility with signatures not fully reduced by the group order. (`pkg/crypto/crypto.go`)
* Added a mainnet test case for Ed25519 verification and improved test structure for better coverage and clarity. (`pkg/crypto/crypto_test.go`) [[1]](diffhunk://#diff-2d1b99bd71bdd031526a498a4b795bfbd6509a5f1b1eeb9ebdbd787ff1b012c9R144) [[2]](diffhunk://#diff-2d1b99bd71bdd031526a498a4b795bfbd6509a5f1b1eeb9ebdbd787ff1b012c9R154-R158) [[3]](diffhunk://#diff-2d1b99bd71bdd031526a498a4b795bfbd6509a5f1b1eeb9ebdbd787ff1b012c9R172)

**Other improvements and code quality:**

* Added a linter exclusion for `pkg/state/exclusions.go` and performed minor import ordering and code cleanup. (`.golangci-strict.yml`, `pkg/crypto/crypto.go`, `pkg/proto/eth_crypto.go`, `pkg/proto/protobuf_converters.go`, `pkg/ride/diff_state.go`, `pkg/ride/environment.go`) [[1]](diffhunk://#diff-ea785ed4d4c52f1e6236726d48b839488a705bf2c1376fec9920091c68740b91R137) [[2]](diffhunk://#diff-8a03969770d7b59692a408c8f8cfcff4c2b4f21281ac81f36d94d4b39306c434L17-R20) [[3]](diffhunk://#diff-0f1c0c4cc6c543505cf001a94c5e127bef2061ad43048da061d79f81090b6f9fR9) [[4]](diffhunk://#diff-39cc270751d0512791f183a74d580f49cc6a7cf557d2b1b53c3e7ccf4efd801cR4-R5) [[5]](diffhunk://#diff-2d59de9747d4808a8d852bc95ad5c37626e936df99094f7564d8b709f7cce670R5) [[6]](diffhunk://#diff-1de2af74b6b93e35c1dbc241b7d233237810665377ead50d6021baa4babebb99R5-R9) [[7]](diffhunk://#diff-2d59de9747d4808a8d852bc95ad5c37626e936df99094f7564d8b709f7cce670R205-R217)

These changes address critical edge cases in signature processing, improve test coverage, and increase the reliability and maintainability of the codebase.

### CODE REVIEW

File: pkg/settings/features.go [hunk 1]
```diff
@@ -29,7 +29,8 @@ const (
 	LightNode               // 22
 	BoostBlockReward        // 23
 	ecrecoverFix            // 24, intentionally package private
-	InvokeExpression        // 25
+	DeterministicFinality   // 25 Deterministic Finality and RideV9
+	InvokeExpression        // 26
```
[the remark this answers is not in the document]
Comment (nickeskov) [reply pkg/settings/features.go:33]: `InvokeExpression` and `DeterministicFinality` marked as not implemented, so they can't affect state database.

File: pkg/state/exclusions.go [hunk 2]
```diff
@@ -42,3 +44,672 @@ func leasesToDisabledAliasesMainnet() []crypto.Digest {
...
+
+func getAbnormalTxMainnet(txID crypto.Digest) (abnormalTxInfo, bool) {
+	abnormalTxsMainnetInitializer.Do(func() { abnormalTxsMainnet = newAbnormalTxsMainnet() })
+	tx, ok := abnormalTxsMainnet[txID]
+	return tx, ok
+}
+
+func cleanAbnormalTxsMainnet() {
+	abnormalTxsMainnetCleaner.Do(func() { abnormalTxsMainnet = nil })
+}
```
[the remark this answers is not in the document]
Comment (nickeskov) [reply pkg/state/exclusions.go:82]: Good point.

**Review by alexeykiselev — APPROVED** [review]
